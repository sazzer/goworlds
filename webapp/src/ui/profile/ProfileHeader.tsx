import React, {FunctionComponent} from 'react';
import {Header} from "semantic-ui-react";
import {NamedUser} from "./NamedUser";

/** The type for the header */
export declare type ProfileHeaderProps = {
    user: NamedUser
};

/**
 * Header to render on the Profile page
 * @constructor
 */
export const ProfileHeader: FunctionComponent<ProfileHeaderProps> = ({user}) => {
    return <Header size='large' dividing>
        {user.name}
    </Header>;
}
