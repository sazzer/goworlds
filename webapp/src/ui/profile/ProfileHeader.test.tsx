import React from 'react';
import {ProfileHeader} from './ProfileHeader';
import {render} from '@testing-library/react';

jest.mock('../common/Authenticated', () => {
    const Authenticated = jest.fn(() => 'Mock Authenticated');
    return {
        Authenticated,
    };
});

/** Set up the component to test */
function setup(name: string = 'Username') {
    return {
        element: render(<ProfileHeader user={{name}} />)
    };
}

it('renders without crashing', () => {
    const {element} = setup();

    expect(element.baseElement).toMatchSnapshot();
});
