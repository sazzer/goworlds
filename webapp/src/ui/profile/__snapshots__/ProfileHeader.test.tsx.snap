// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<body>
  <div>
    <div
      class="ui large dividing header"
    >
      Username
    </div>
  </div>
</body>
`;
